src/App.tsx

```diff
@@ -15,36 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+const handleSort = (
+  type: SortType,
+  setGoods: React.Dispatch<React.SetStateAction<string[]>>,
+  isReversed: boolean,
+) => {
+  let sortedGoods = [...goodsFromServer];
+
+  switch (type) {
+    case SortType.ALPHABET:
+      sortedGoods.sort();
+      break;
+    case SortType.LENGTH:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goodsFromServer];
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  setGoods(sortedGoods);
+};
+
+const handleReverse = (
```

Comment:
- You reverse goods in `handleSort`, so you don't need this one
- You don't need update the goods state. It could be just a variable

Reply:
Not fixed!